doc/compiling/README.md

```diff
@@ -23,6 +23,7 @@ General options and their default values:
     PRECOMPILE_HEADERS=FALSE   - Precompile some headers (experimental; requires CMake 3.16 or later)
     PRECOMPILED_HEADERS_PATH=  - Path to a file listing all headers to precompile (default points to src/precompiled_headers.txt)
     USE_SDL2=TRUE              - Build with SDL2; Enables IrrlichtMt device SDL2
+    USE_SDL2_STATIC=TRUE       - Links with SDL2::SDL2-static instead of SDL2::SDL2
```

Comment:
is this required on macOS for some reason? if so, why?

Reply:
I am using static linking on macOS because it makes application signing much easier, and I can generate an Xcode project from CMake and do not need to keep the Xcode project file in the repository, and keep it up to date with changes in CMake manually.
By my findings, if SDL is built with shared and static at the same time, users have to use SDL2::SDL2-static instead of SDL2::SDL2 to link statically.

Reply:
hmm. since you're already passing `SDL2_LIBRARIES` explicitly, doesn't that work as-is?

Reply:
Hm. Is residuum. I forgot to remove.
It is copied from Android style, but it will require some exceptions like this to be useful:
https://github.com/luanti-org/luanti/blob/01e439597730371f0ff2f9d89141c05c74f81597/irr/src/CMakeLists.txt#L231-L236
because the behaviors I see were overriding `SDL1_LIBRARIES` by `find_package(SDL2...` command